packages/catalog-process/src/services/CatalogService.ts

```diff
@@ -227,4 +237,124 @@ export const catalogService = {
       },
     });
   },
+
+  async createDescriptor(
+    eServiceId: string,
+    eserviceDescriptorSeed: EServiceDescriptorSeed
+  ): Promise<string> {
+    const eservice = await readModelGateway.getEServiceById(eServiceId);
+    if (eservice === undefined) {
+      throw eServiceNotFound(eServiceId);
+    }
+
+    const newVersion = nextDescriptorVersion(eservice);
+    const descriptorId = uuidv4();
+    const createCatalogDescriptor = descriptorSeedToCreateEvent(
+      descriptorId,
+      eserviceDescriptorSeed,
+      newVersion.toString()
+    );
+
+    await eventRepository.createEvent(createCatalogDescriptor);
+    return descriptorId;
+  },
+
+  async deleteDraftDescriptor(
+    eServiceId: string,
+    descriptorId: string
+  ): Promise<void> {
```

Comment:
Same as before, are we missing the check from the catalog process api?

Reply:
🙏🏻